vcpkg features

renefritze opened the pull request.

Commit:
remove uggrid

currently does not support the cmake superbuild stuff:
CMake Error at deps/dune-common/cmake/modules/DuneAddLibrary.cmake:179
(add_library):
add_library cannot create target "duneuggrid" because an imported target
with the same name already exists.
Call Stack (most recent call first):
deps/dune-common/cmake/modules/DuneAddLibrary.cmake:148
(dune_add_library_normal)
deps/dune-uggrid/CMakeLists.txt:66 (dune_add_library)

CMake Error at deps/dune-uggrid/CMakeLists.txt:68
(target_link_libraries):
INTERFACE library can only be used with the INTERFACE keyword of
target_link_libraries

CMake Error at deps/dune-uggrid/dune/uggrid/CMakeLists.txt:15
(target_sources):
target_sources may only set INTERFACE properties on IMPORTED targets

CMake Error at deps/dune-uggrid/dune/uggrid/CMakeLists.txt:15
(target_sources):
target_sources may only set INTERFACE properties on IMPORTED targets

CMake Error at deps/dune-uggrid/dune/uggrid/CMakeLists.txt:27
(target_sources):
target_sources may only set INTERFACE properties on IMPORTED targets

CMake Error at deps/dune-uggrid/dune/uggrid/low/CMakeLists.txt:6
(target_sources):
target_sources may only set INTERFACE properties on IMPORTED targets

CMake Error at
deps/dune-uggrid/dune/uggrid/parallel/ddd/CMakeLists.txt:27
(target_sources):
target_sources may only set INTERFACE properties on IMPORTED targets

CMake Error at
deps/dune-uggrid/dune/uggrid/parallel/ppif/CMakeLists.txt:8
(target_sources):
target_sources may only set INTERFACE properties on IMPORTED targets

Commit:
cmake| switch to superbuild

Commit:
switch submodule urls to "origina" upstreams for 2.10 tags

Commit:
adds a dummy preset set

Commit:
vcpkg setup

Commit:
remmove uggrid and alugrid

both throw cmake errors on generate

Commit:
fix testtools code generation

Commit:
add missing boost packages

Commit:
fix boost ofstream not defined

Commit:
necessary CollectiveCommunication -> Communication changes

Commit:
fix some deprecated getCollectiveCommunication calls

Commit:
update submodules

Commit:
disable tests that require simplicial grids

Commit:
fix interpolation

Commit:
try diff quadrature point iteration

1) new dune-geometry supports getting a structured bindings return from
::rule
2) `auto&&` was an effort to try to remove dangling ref warnings

both modes show this warning now though, and I can't follow _why_

Commit:
fmatrix: disable operator* overloads

we had these cuase upstream matrix type didn't define them (I think)

Commit:
disable vijayasundaram code

it leads to multiple errors in vector/matrix math

Commit:
random| fix a c++20 warning

Commit:
switch dune-common to fork

Commit:
udpate pre-commit hook versions

Commit:
spply ruff update

Commit:
ci| adds a simple non-docker workflow

Commit:
fix headercheck

Commit:
add a unique label to our own tests and filter ctest execution on it

Commit:
fix most test running

Commit:
add overlay ports

Commit:
update vcpkg infra

Commit:
cmake changes for vcpkg setup

Commit:
update testtools

Commit:
adds a vcpkg port sync script

Commit:
update ports

Commit:
remove grid glue + uggrid

Commit:
upload logs

Commit:
upload logs

Commit:
update overlay ports with dependencies

Commit:
adds an alberta port

Commit:
cleanup cmake list

Commit:
fix alberta finding libtirpc

Commit:
fix linking alberta

Commit:
try limiting load via ninja

Commit:
remove obsolete merge script

Commit:
alberta doesn't build in manylinux out-of-the-box

these changes would get us closer, if the patch wasn't deemed corrupt by
vcpkg

Commit:
rewire wheel-building

- use vanilla manylinux image
- no more dunecontrol

Commit:
make libtirp linking opnly required when alberta was found

Commit:
update pybind11 2.10 -> 2.13

Commit:
fix pre-commit run in update_ports

Commit:
install missing tools in manylinux container

Commit:
update pybind11 port

Commit:
actually use pybind11 from vcpkg

Commit:
fix wheel builder script

Commit:
remove unused make_wheels script

Commit:
extend pre-commit setup for GHA validation

Commit:
pathlib for make_env_file

Commit:
ci| add wheel building

Commit:
fix test.pypi clean up script

Commit:
introduce optional deps for alberta+ug

Commit:
adds preset for wheelbuilders that doesn't enable alberta and UG

Commit:
cleanup presets

Commit:
prevent multiple output wheels

Commit:
remove alberta port patching

doesn't work outside manylinux container

Commit:
remove almost all submodules

instead keep a manual list of modules hash+url and source that in the
port generation script
then clone those repos and extract version info if possible

ftschindler commented on .vcpkg-overlays/ports/dune-alugrid/vcpkg.json:
```diff
@@ -15,5 +15,15 @@
-  ]
+  ],
+  "features": {
+    "uggrid": {
```
Why is UG a feature of dune-alugrid?

renefritze replied:
Stuff breaks otherwise :)
It's an order thing, if alugrid is configured before UG, then stuff breaks. 
I _think_ this might be due to `dune.module` parsing and suggested deps there.

renefritze committed:
fix typo

Co-authored-by: Felix Schindler <107669806+ftschindler@users.noreply.github.com>

Commit:
fix deprecated function use

Commit:
remove broken yapf config links

Commit:
cleanup cmake

Commit:
fix venv creation

Commit:
fix port dependency order

Commit:
disable MPI findings

Commit:
put MPI disabling into the portfiles

Commit:
add suitesparse + openblas

Commit:
some more optional deps

Commit:
workaround saddlepoint solver problem in dune-istl

disable the possibility to use ISTL with the saddlepoint solver
altogether. Move files accordingly and remove specialized tests
